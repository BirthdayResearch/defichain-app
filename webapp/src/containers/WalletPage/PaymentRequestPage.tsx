import React, { Component } from "react";
import { Helmet } from "react-helmet";
import { Button, ButtonGroup } from "reactstrap";
import { MdArrowBack, MdDelete } from "react-icons/md";
import { NavLink, RouteComponentProps } from "react-router-dom";
import KeyValueLi from "../../components/KeyValueLi/KeyValueLi";
import { I18n } from "react-redux-i18n";

interface PaymentRequestPageProps {}

interface PaymentRequestPageState {
  label: string;
  amount: string;
  time: string;
  message: string;
  address: string;
}

interface RouteProps {
  id: string;
}

class PaymentRequestPage extends Component<
  PaymentRequestPageProps & RouteComponentProps<RouteProps>,
  PaymentRequestPageState
> {
  state = {
    label: "TREQ 9876",
    amount: "1.5",
    time: "Feb 19, 2:03 pm",
    message: "Ref ID REQ456789",
    address: "bc1qnckyj0jxrcgtu4j90r0efcae750rfx6555rhaq",
  };

  render() {
    return (
      <div className="main-wrapper">
        <Helmet>
          <title>
            {I18n.t(
              "containers.wallet.paymentRequestPage.paymentRequestTitle",
              {
                id: this.props.match.params.id,
              }
            )}
          </title>
        </Helmet>
        <header className="header-bar">
          <Button to="/" tag={NavLink} color="link" className="header-bar-back">
            <MdArrowBack />
            <span className="d-lg-inline">
              {I18n.t("containers.wallet.paymentRequestPage.wallet")}
            </span>
          </Button>
          <h1>
            {I18n.t("containers.wallet.paymentRequestPage.paymentRequest")}
            &nbsp;
            {this.props.match.params.id}
          </h1>
          <ButtonGroup>
            <Button color="link">
              <MdDelete />
              <span>
                {I18n.t("containers.wallet.paymentRequestPage.delete")}
              </span>
            </Button>
          </ButtonGroup>
        </header>
        <div className="content">
          <section className="mb-5">
            <KeyValueLi
              label={I18n.t("containers.wallet.paymentRequestPage.label")}
              value={this.state.label}
            />
            <KeyValueLi
              label={I18n.t("containers.wallet.paymentRequestPage.amount")}
              value={this.state.amount}
            />
            <KeyValueLi
              label={I18n.t("containers.wallet.paymentRequestPage.time")}
              value={this.state.time}
            />
            <KeyValueLi
              label={I18n.t("containers.wallet.paymentRequestPage.message")}
              value={this.state.message}
            />
            <KeyValueLi
              label={I18n.t("containers.wallet.paymentRequestPage.address")}
              value={this.state.address}
              popsQR={true}
              copyable={true!}
              uid="address"
            />
            <KeyValueLi
              label={I18n.t("containers.wallet.paymentRequestPage.uRI")}
              value={`
              bitcoin:${this.state.address}?amount=${this.state.amount}&label=${this.state.label}&message=${this.state.message}
            `}
              popsQR={true}
              copyable={true}
              uid="uri"
            />
          </section>
        </div>
      </div>
    );
  }
}

export default PaymentRequestPage;
