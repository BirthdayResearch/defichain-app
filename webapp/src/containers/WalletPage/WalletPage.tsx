import React, { Component } from "react";
import { Helmet } from "react-helmet";
import { Button, ButtonGroup, Row, Col } from "reactstrap";
import { MdArrowUpward, MdArrowDownward } from "react-icons/md";
import { NavLink as RRNavLink } from "react-router-dom";
import StatCard from "../../components/StatCard/StatCard";
import WalletTxns from "./WalletTxns";
import PaymentRequests from "./PaymentRequests";
import { I18n } from "react-redux-i18n";
import { connect } from "react-redux";
import { fetchWalletBalanceRequest } from "./reducer";

interface WalletPageProps {
  walletBalance: string;
  fetchWalletBalance: Function;
}

interface WalletPageState {
  activeTab: string;
}

class WalletPage extends Component<WalletPageProps, WalletPageState> {
  state = {
    activeTab: "txns",
  };

  componentDidMount() {
    this.props.fetchWalletBalance();
  }

  setActiveTab = (tab: string) => {
    if (this.state.activeTab !== tab) {
      this.setState({
        activeTab: tab,
      });
    }
  };

  render() {
    const { walletBalance } = this.props;
    return (
      <div className="main-wrapper">
        <Helmet>
          <title>
            {I18n.t("containers.wallet.walletPage.walletDefiClient")}
          </title>
        </Helmet>
        <header className="header-bar">
          <h1>{I18n.t("containers.wallet.walletPage.wallet")}</h1>
          <ButtonGroup>
            <Button to="/wallet/send" tag={RRNavLink} color="link" size="sm">
              <MdArrowUpward />
              <span className="d-md-inline">
                {I18n.t("containers.wallet.walletPage.send")}
              </span>
            </Button>
            <Button to="/wallet/receive" tag={RRNavLink} color="link" size="sm">
              <MdArrowDownward />
              <span className="d-md-inline">
                {I18n.t("containers.wallet.walletPage.receive")}
              </span>
            </Button>
          </ButtonGroup>
        </header>
        <div className="content">
          <section>
            <Row>
              <Col>
                <StatCard
                  label={I18n.t(
                    "containers.wallet.walletPage.availableBalance"
                  )}
                  value={walletBalance}
                  unit="DFI"
                />
              </Col>
              <Col>
                <StatCard
                  label={I18n.t("containers.wallet.walletPage.pending")}
                  value="1,000"
                  unit="DFI"
                />
              </Col>
            </Row>
          </section>
          <section>
            <h2>{I18n.t("containers.wallet.walletPage.paymentRequests")}</h2>
            <PaymentRequests />
          </section>
          <section>
            <h2>{I18n.t("containers.wallet.walletPage.transactions")}</h2>
            <WalletTxns />
          </section>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { walletBalance } = state.wallet;
  return {
    walletBalance,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    fetchWalletBalance: () => dispatch(fetchWalletBalanceRequest()),
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(WalletPage);
