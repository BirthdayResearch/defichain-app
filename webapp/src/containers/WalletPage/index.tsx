import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';
import { I18n } from 'react-redux-i18n';
import { Button, ButtonGroup, Row, Col, Badge } from 'reactstrap';
import {
  MdArrowUpward,
  MdArrowDownward,
  MdRefresh,
  MdArrowBack,
} from 'react-icons/md';

import { NavLink as RRNavLink, RouteComponentProps } from 'react-router-dom';
import StatCard from '../../components/StatCard';
import WalletTxns from './components/WalletTxnsNew';
import {
  fetchInstantBalanceRequest,
  fetchInstantPendingBalanceRequest,
} from './reducer';
import {
  BTC_SPV_SYMBOL,
  DFI_SYMBOL,
  DST,
  LP,
  WALLET_TOKENS_PATH,
} from '../../constants';
import { startUpdateApp, openBackupWallet } from '../PopOver/reducer';
import { WALLET_SEND_PATH, WALLET_RECEIVE_PATH } from '../../constants';
import {
  getPageTitle,
  getSymbolKey,
  getTokenDisplayName,
} from '../../utils/utility';
import styles from './WalletPage.module.scss';
import TokenAvatar from '../../components/TokenAvatar';
import Header from '../HeaderComponent';
import { WalletPathEnum } from './types';
import { getWalletPathAddress } from './service';

interface WalletPageProps extends RouteComponentProps {
  unit: string;
  walletBalance: string;
  pendingBalance: string;
  fetchInstantBalanceRequest: () => void;
  fetchInstantPendingBalanceRequest: () => void;
  blockChainInfo: any;
}

const WalletPage: React.FunctionComponent<WalletPageProps> = (
  props: WalletPageProps
) => {
  const urlParams = new URLSearchParams(props.location.search);
  const tokenSymbol = urlParams.get(WalletPathEnum.symbol);
  const tokenHash = urlParams.get(WalletPathEnum.hash) || '';
  const tokenAmount = urlParams.get(WalletPathEnum.amount);
  const tokenAddress = urlParams.get(WalletPathEnum.address);
  const isLPS = urlParams.get(WalletPathEnum.isLPS) == 'true';
  const isSPV = urlParams.get(WalletPathEnum.isSPV) == 'true';
  const displayName = urlParams.get(WalletPathEnum.displayName) || '';
  const isDAT = urlParams.get(WalletPathEnum.isDAT) == 'true';

  const {
    fetchInstantBalanceRequest,
    unit,
    fetchInstantPendingBalanceRequest,
    walletBalance,
    pendingBalance,
  } = props;

  useEffect(() => {
    fetchInstantBalanceRequest();
    fetchInstantPendingBalanceRequest();

    return () => {
      clearTimeout(balanceRefreshTimerID);
      clearTimeout(pendingBalRefreshTimerID);
    };
  }, []);

  let balanceRefreshTimerID;
  let pendingBalRefreshTimerID;
  const [refreshBalance, setRefreshBalance] = useState(false);
  const [pendingRefreshBalance, setPendingRefreshBalance] = useState(false);

  return (
    <div className='main-wrapper'>
      <Helmet>
        <title>
          {getPageTitle(I18n.t('containers.wallet.walletPage.walletDeFiApp'))}
        </title>
      </Helmet>
      <Header>
        <Button
          to={`${WALLET_TOKENS_PATH}?value=${walletBalance}&unit=${unit}`}
          tag={RRNavLink}
          color='link'
          className='header-bar-back'
        >
          <MdArrowBack />
          <span className='d-lg-inline'>
            {I18n.t('containers.wallet.walletPage.balances')}
          </span>
        </Button>
        <div className={styles.titleWithIcon}>
          <TokenAvatar
            symbol={
              tokenSymbol
                ? getTokenDisplayName(
                    isDAT,
                    tokenSymbol,
                    tokenSymbol,
                    tokenHash,
                    isSPV
                  )
                : unit
            }
          />
          <h1 className='d-flex align-items-center'>
            {tokenSymbol
              ? getTokenDisplayName(
                  isDAT,
                  displayName,
                  tokenSymbol,
                  tokenHash,
                  isSPV
                )
              : unit}{' '}
            {I18n.t('containers.wallet.walletPage.wallet')}
            {![DFI_SYMBOL, BTC_SPV_SYMBOL].includes(tokenHash ?? '') && (
              <Badge className='ml-2' color='disabled'>
                {isLPS ? LP : DST}
              </Badge>
            )}
          </h1>
        </div>
        <ButtonGroup>
          <Button
            to={
              tokenSymbol
                ? getWalletPathAddress(
                    WALLET_SEND_PATH,
                    tokenSymbol,
                    tokenHash || '',
                    tokenAmount || '',
                    tokenAddress || '',
                    isLPS,
                    isSPV,
                    displayName || '',
                    isDAT
                  )
                : WALLET_SEND_PATH
            }
            tag={RRNavLink}
            color='link'
            size='sm'
          >
            <MdArrowUpward />
            <span className='d-md-inline'>
              {I18n.t('containers.wallet.walletPage.send')}
            </span>
          </Button>
          <Button
            to={
              tokenSymbol
                ? getWalletPathAddress(
                    WALLET_RECEIVE_PATH,
                    tokenSymbol,
                    tokenHash || '',
                    tokenAmount || '',
                    tokenAddress || '',
                    isLPS,
                    isSPV,
                    displayName || '',
                    isDAT
                  )
                : WALLET_RECEIVE_PATH
            }
            tag={RRNavLink}
            color='link'
            size='sm'
          >
            <MdArrowDownward />
            <span className='d-md-inline'>
              {I18n.t('containers.wallet.walletPage.receive')}
            </span>
          </Button>
        </ButtonGroup>
      </Header>
      <div className='content'>
        <section>
          <Row>
            <Col>
              <StatCard
                label={I18n.t('containers.wallet.walletPage.availableBalance')}
                value={tokenAmount ? tokenAmount : walletBalance}
                unit={
                  tokenSymbol
                    ? getTokenDisplayName(isDAT, displayName, tokenSymbol, tokenHash, isSPV)
                    : unit
                }
                refreshFlag={refreshBalance}
                icon={
                  <MdRefresh
                    className={styles.iconPointer}
                    size={30}
                    onClick={() => {
                      setRefreshBalance(true);
                      balanceRefreshTimerID = setTimeout(() => {
                        setRefreshBalance(false);
                      }, 2000);
                      fetchInstantBalanceRequest();
                    }}
                  />
                }
              />
            </Col>
            {/* <Col>
              <StatCard
                label={I18n.t('containers.wallet.walletPage.pending')}
                value={getAmountInSelectedUnit(pendingBalance, unit)}
                unit={
                  tokenSymbol
                    ? getSymbolKey(tokenSymbol, tokenHash || DFI_SYMBOL)
                    : unit
                }
                refreshFlag={pendingRefreshBalance}
                icon={
                  <MdRefresh
                    className={styles.iconPointer}
                    size={30}
                    onClick={() => {
                      setPendingRefreshBalance(true);
                      pendingBalRefreshTimerID = setTimeout(() => {
                        setPendingRefreshBalance(false);
                      }, 2000);
                      fetchInstantPendingBalanceRequest();
                    }}
                  />
                }
              />
            </Col> */}
          </Row>
        </section>
        <WalletTxns
          tokenSymbol={
            tokenSymbol
              ? getSymbolKey(tokenSymbol, tokenHash || DFI_SYMBOL, isLPS)
              : unit
          }
        />
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const {
    wallet: { walletBalance, pendingBalance, blockChainInfo },
    settings: {
      appConfig: { unit },
    },
  } = state;
  return {
    unit,
    walletBalance,
    pendingBalance,
    blockChainInfo,
  };
};

const mapDispatchToProps = {
  fetchInstantBalanceRequest,
  fetchInstantPendingBalanceRequest,
  startUpdateApp,
  openBackupWallet,
};

export default connect(mapStateToProps, mapDispatchToProps)(WalletPage);
