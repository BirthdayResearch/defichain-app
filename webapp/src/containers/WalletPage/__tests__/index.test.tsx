import React from 'react';
import WalletPage from '../index';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../app/rootStore';
import { mount } from 'enzyme';

describe('WalletPage component', () => {
  it('should check for snapshot', () => {
    const wrapper = mount(
      <Router>
        <Provider store={store}>
          <WalletPage />
        </Provider>
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
