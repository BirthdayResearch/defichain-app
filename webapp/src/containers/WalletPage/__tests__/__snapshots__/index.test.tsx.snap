// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletPage component should check for snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(WalletPage)>
        <WalletPage
          fetchPendingBalanceRequest={[Function]}
          fetchWalletBalanceRequest={[Function]}
          openBackupWallet={[Function]}
          pendingBalance={0}
          startUpdateApp={[Function]}
          unit="DFI"
          updateAvailableBadge={false}
          walletBalance={0}
        >
          <div
            className="main-wrapper"
          >
            <HelmetWrapper
              defer={true}
              encodeSpecialCharacters={true}
            >
              <SideEffect(NullComponent)
                defer={true}
                encodeSpecialCharacters={true}
                title="Wallet – DeFi Blockchain Client"
                titleAttributes={Object {}}
              >
                <NullComponent
                  defer={true}
                  encodeSpecialCharacters={true}
                  title="Wallet – DeFi Blockchain Client"
                  titleAttributes={Object {}}
                />
              </SideEffect(NullComponent)>
            </HelmetWrapper>
            <header
              className="header-bar"
            >
              <h1>
                Wallet
              </h1>
              <ButtonGroup
                role="group"
                tag="div"
              >
                <div
                  className="btn-group"
                  role="group"
                >
                  <Button
                    color="link"
                    size="sm"
                    tag={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "NavLink",
                        "propTypes": Object {
                          "activeClassName": [Function],
                          "activeStyle": [Function],
                          "aria-current": [Function],
                          "className": [Function],
                          "exact": [Function],
                          "innerRef": [Function],
                          "isActive": [Function],
                          "location": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "strict": [Function],
                          "style": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    to="/wallet/send"
                  >
                    <NavLink
                      aria-label={null}
                      className="btn btn-link btn-sm"
                      onClick={[Function]}
                      to="/wallet/send"
                    >
                      <Link
                        aria-current={null}
                        aria-label={null}
                        className="btn btn-link btn-sm"
                        onClick={[Function]}
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/wallet/send",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          aria-label={null}
                          className="btn btn-link btn-sm"
                          href="/wallet/send"
                          navigate={[Function]}
                          onClick={[Function]}
                        >
                          <a
                            aria-current={null}
                            aria-label={null}
                            className="btn btn-link btn-sm"
                            href="/wallet/send"
                            onClick={[Function]}
                          >
                            <MdArrowUpward>
                              <IconBase
                                attr={
                                  Object {
                                    "viewBox": "0 0 24 24",
                                  }
                                }
                              >
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  stroke="currentColor"
                                  strokeWidth="0"
                                  style={
                                    Object {
                                      "color": undefined,
                                    }
                                  }
                                  viewBox="0 0 24 24"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                                    key="0"
                                  />
                                </svg>
                              </IconBase>
                            </MdArrowUpward>
                            <span
                              className="d-md-inline"
                            >
                              Send
                            </span>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                  </Button>
                  <Button
                    color="link"
                    size="sm"
                    tag={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "NavLink",
                        "propTypes": Object {
                          "activeClassName": [Function],
                          "activeStyle": [Function],
                          "aria-current": [Function],
                          "className": [Function],
                          "exact": [Function],
                          "innerRef": [Function],
                          "isActive": [Function],
                          "location": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "strict": [Function],
                          "style": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    to="/wallet/receive"
                  >
                    <NavLink
                      aria-label={null}
                      className="btn btn-link btn-sm"
                      onClick={[Function]}
                      to="/wallet/receive"
                    >
                      <Link
                        aria-current={null}
                        aria-label={null}
                        className="btn btn-link btn-sm"
                        onClick={[Function]}
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/wallet/receive",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          aria-label={null}
                          className="btn btn-link btn-sm"
                          href="/wallet/receive"
                          navigate={[Function]}
                          onClick={[Function]}
                        >
                          <a
                            aria-current={null}
                            aria-label={null}
                            className="btn btn-link btn-sm"
                            href="/wallet/receive"
                            onClick={[Function]}
                          >
                            <MdArrowDownward>
                              <IconBase
                                attr={
                                  Object {
                                    "viewBox": "0 0 24 24",
                                  }
                                }
                              >
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  stroke="currentColor"
                                  strokeWidth="0"
                                  style={
                                    Object {
                                      "color": undefined,
                                    }
                                  }
                                  viewBox="0 0 24 24"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                                    key="0"
                                  />
                                </svg>
                              </IconBase>
                            </MdArrowDownward>
                            <span
                              className="d-md-inline"
                            >
                              Receive
                            </span>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                  </Button>
                </div>
              </ButtonGroup>
            </header>
            <div
              className="content"
            >
              <section>
                <Row
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <div
                    className="row"
                  >
                    <Col
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col"
                      >
                        <StatCard
                          icon={
                            <MdRefresh
                              onClick={[Function]}
                              size={30}
                            />
                          }
                          label="Available balance"
                          refreshFlag={false}
                          unit="DFI"
                          value="0"
                        >
                          <Card
                            tag="div"
                          >
                            <div
                              className="card"
                            >
                              <CardBody
                                tag="div"
                              >
                                <div
                                  className="card-body"
                                >
                                  <Row
                                    tag="div"
                                    widths={
                                      Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                                    }
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        tag="div"
                                        widths={
                                          Array [
                                            "xs",
                                            "sm",
                                            "md",
                                            "lg",
                                            "xl",
                                          ]
                                        }
                                      >
                                        <div
                                          className="col"
                                        >
                                          Available balance
                                        </div>
                                      </Col>
                                      <Col
                                        className="undefined "
                                        md="6"
                                        tag="div"
                                        widths={
                                          Array [
                                            "xs",
                                            "sm",
                                            "md",
                                            "lg",
                                            "xl",
                                          ]
                                        }
                                      >
                                        <div
                                          className="undefined  col-md-6"
                                        >
                                          <MdRefresh
                                            onClick={[Function]}
                                            size={30}
                                          >
                                            <IconBase
                                              attr={
                                                Object {
                                                  "viewBox": "0 0 24 24",
                                                }
                                              }
                                              onClick={[Function]}
                                              size={30}
                                            >
                                              <svg
                                                fill="currentColor"
                                                height={30}
                                                onClick={[Function]}
                                                size={30}
                                                stroke="currentColor"
                                                strokeWidth="0"
                                                style={
                                                  Object {
                                                    "color": undefined,
                                                  }
                                                }
                                                viewBox="0 0 24 24"
                                                width={30}
                                                xmlns="http://www.w3.org/2000/svg"
                                              >
                                                <path
                                                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                                                  key="0"
                                                />
                                              </svg>
                                            </IconBase>
                                          </MdRefresh>
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                  <Row
                                    tag="div"
                                    widths={
                                      Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                                    }
                                  >
                                    <div
                                      className="row"
                                    >
                                       
                                      <Col
                                        tag="div"
                                        widths={
                                          Array [
                                            "xs",
                                            "sm",
                                            "md",
                                            "lg",
                                            "xl",
                                          ]
                                        }
                                      >
                                        <div
                                          className="col"
                                        >
                                          0
                                        </div>
                                      </Col>
                                      <Col
                                        style={
                                          Object {
                                            "lineHeight": "2.5rem",
                                            "textAlign": "end",
                                          }
                                        }
                                        tag="div"
                                        widths={
                                          Array [
                                            "xs",
                                            "sm",
                                            "md",
                                            "lg",
                                            "xl",
                                          ]
                                        }
                                      >
                                        <div
                                          className="col"
                                          style={
                                            Object {
                                              "lineHeight": "2.5rem",
                                              "textAlign": "end",
                                            }
                                          }
                                        >
                                          DFI
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                </div>
                              </CardBody>
                            </div>
                          </Card>
                        </StatCard>
                      </div>
                    </Col>
                    <Col
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col"
                      >
                        <StatCard
                          icon={
                            <MdRefresh
                              onClick={[Function]}
                              size={30}
                            />
                          }
                          label="Pending"
                          refreshFlag={false}
                          unit="DFI"
                          value="0"
                        >
                          <Card
                            tag="div"
                          >
                            <div
                              className="card"
                            >
                              <CardBody
                                tag="div"
                              >
                                <div
                                  className="card-body"
                                >
                                  <Row
                                    tag="div"
                                    widths={
                                      Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                                    }
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        tag="div"
                                        widths={
                                          Array [
                                            "xs",
                                            "sm",
                                            "md",
                                            "lg",
                                            "xl",
                                          ]
                                        }
                                      >
                                        <div
                                          className="col"
                                        >
                                          Pending
                                        </div>
                                      </Col>
                                      <Col
                                        className="undefined "
                                        md="6"
                                        tag="div"
                                        widths={
                                          Array [
                                            "xs",
                                            "sm",
                                            "md",
                                            "lg",
                                            "xl",
                                          ]
                                        }
                                      >
                                        <div
                                          className="undefined  col-md-6"
                                        >
                                          <MdRefresh
                                            onClick={[Function]}
                                            size={30}
                                          >
                                            <IconBase
                                              attr={
                                                Object {
                                                  "viewBox": "0 0 24 24",
                                                }
                                              }
                                              onClick={[Function]}
                                              size={30}
                                            >
                                              <svg
                                                fill="currentColor"
                                                height={30}
                                                onClick={[Function]}
                                                size={30}
                                                stroke="currentColor"
                                                strokeWidth="0"
                                                style={
                                                  Object {
                                                    "color": undefined,
                                                  }
                                                }
                                                viewBox="0 0 24 24"
                                                width={30}
                                                xmlns="http://www.w3.org/2000/svg"
                                              >
                                                <path
                                                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                                                  key="0"
                                                />
                                              </svg>
                                            </IconBase>
                                          </MdRefresh>
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                  <Row
                                    tag="div"
                                    widths={
                                      Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                                    }
                                  >
                                    <div
                                      className="row"
                                    >
                                       
                                      <Col
                                        tag="div"
                                        widths={
                                          Array [
                                            "xs",
                                            "sm",
                                            "md",
                                            "lg",
                                            "xl",
                                          ]
                                        }
                                      >
                                        <div
                                          className="col"
                                        >
                                          0
                                        </div>
                                      </Col>
                                      <Col
                                        style={
                                          Object {
                                            "lineHeight": "2.5rem",
                                            "textAlign": "end",
                                          }
                                        }
                                        tag="div"
                                        widths={
                                          Array [
                                            "xs",
                                            "sm",
                                            "md",
                                            "lg",
                                            "xl",
                                          ]
                                        }
                                      >
                                        <div
                                          className="col"
                                          style={
                                            Object {
                                              "lineHeight": "2.5rem",
                                              "textAlign": "end",
                                            }
                                          }
                                        >
                                          DFI
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                </div>
                              </CardBody>
                            </div>
                          </Card>
                        </StatCard>
                      </div>
                    </Col>
                  </div>
                </Row>
              </section>
              <Connect(WalletTxns)>
                <WalletTxns
                  fetchWalletTxns={[Function]}
                  stopPagination={true}
                  unit="DFI"
                  walletTxnCount={0}
                  walletTxns={Array []}
                >
                  <section
                    className="mb-5"
                  >
                    <h2>
                      Transactions
                    </h2>
                    <Card
                      className="table-responsive-md"
                      tag="div"
                    >
                      <div
                        className="table-responsive-md card"
                      >
                        <CardBody
                          tag="div"
                        >
                          <div
                            className="card-body"
                          >
                            No transactions available
                          </div>
                        </CardBody>
                      </div>
                    </Card>
                  </section>
                </WalletTxns>
              </Connect(WalletTxns)>
            </div>
          </div>
        </WalletPage>
      </Connect(WalletPage)>
    </Provider>
  </Router>
</BrowserRouter>
`;
